src/main/java/com/google/flourbot/datastorage/FirebaseDataStorage.java

```diff
@@ -45,8 +45,8 @@ public FirebaseDataStorage() {
     // Create a query to find a macro named macroName belonging to userEmail
     Query query =
         db.collection(COLLECTION_NAME)
-            .whereEqualTo(USER_IDENTIFIER, userEmail)
-            .whereEqualTo(MACRO_IDENTIFIER, macroName);
+            .whereEqualTo(MACRO_IDENTIFIER, macroName)
+            .whereArrayContains(SCOPE_IDENTIFIER, userEmail);
```

Comment:
I'm not sure if using an index helps or speeds the query up, but worth looking into it (https://firebase.corp.google.com/project/stepladder-2020/database/firestore/indexes).